pkg/apis/pipeline/v1beta1/pipeline_validation.go

```diff
@@ -253,6 +254,23 @@ func validatePipelineContextVariables(tasks []PipelineTask) *apis.FieldError {
 	return errs.Also(validatePipelineContextVariablesInParamValues(paramValues, "context\\.pipeline", pipelineContextNames))
 }
 
+func validateExecutionStatusContextVariables(tasks []PipelineTask) (errs *apis.FieldError) {
+	// creating a list of pipelineTask names to validate context.pipelineRun.Tasks.<name>
+	pipelineRunTasksContextNames := sets.String{}
+	var paramValues []string
+	for _, t := range tasks {
+		pipelineRunTasksContextNames.Insert(t.Name)
+		for _, param := range t.Params {
```

Comment:
I'm a little confused; maybe I'm misreading the code.  I think we want to validate the following:

1. Normal (non-final) tasks cannot use the $(tasks.<pipelinetask>.status) syntax.
2. Final tasks can use this syntax but must refer to a defined pipeline task.

Reading the code, it seems to be validating the parameter references in normal tasks refer to other existing pipeline tasks?

Reply:
Great catch @GregDritschler your are absolutely correct.

This validation checks a pipelineTask specified in `$(tasks.pipelinetask.status)` exists in the pipeline. This function is old and needs to be updated. Will fix it.

Reply:
Fixed 🙏